action.yml

```diff
@@ -14,12 +14,29 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Resolve Golang version and set GOTOOLCHAIN
+      id: version
+      run: |
+        semver="${{ inputs.version }}"
+        if [[ -n "${{ inputs.version-file }}" ]]; then
+          goVersion=$(cat "${{ inputs.version-file }}" | grep --extended-regexp "^go +[0-9]")
+          # ignore PATCH component - only consider MAJOR.MINOR
+          if [[ $goVersion =~ ^go\ +([0-9]\.[0-9]+) ]]; then
+            semver=${BASH_REMATCH[1]}
+          else
+            echo "::error title=Extract Golang version from file::Unable to locate valid semver value from module file: ${{ inputs.version-file }}."
+            exit 1
+          fi
+        fi
```

Comment:
Extract only `MAJOR.MINOR` components from the given `go.mod` file via `inputs.version-file` action input argument.